pkg/inspector/image-inspector.go

```diff
@@ -272,23 +272,37 @@ func generateRandomName() (string, error) {
 	return fmt.Sprintf("image-inspector-%016x", n), nil
 }
 
-func getAuthConfigs(dockercfg, username, password_file string) (*docker.AuthConfigurations, error) {
+func appendDockerCfgConfigs(dockercfg string, cfgs *docker.AuthConfigurations) error {
+	var imagePullAuths *docker.AuthConfigurations
+	reader, err := os.Open(dockercfg)
+	if err != nil {
+		return fmt.Errorf("Unable to open docker config file: %v\n", err)
+	}
+	if imagePullAuths, err = docker.NewAuthConfigurations(reader); err != nil {
+		return fmt.Errorf("Unable to parse docker config file: %v\n", err)
+	}
+	if len(imagePullAuths.Configs) == 0 {
+		return fmt.Errorf("No auths were found in the given dockercfg file\n")
+	}
+	for name, ac := range imagePullAuths.Configs {
+		cfgs.Configs[fmt.Sprintf("%s/%s", dockercfg, name)] = ac
+	}
+	return nil
+}
+
+func getAuthConfigs(dockercfg, username, passwordFile string) (*docker.AuthConfigurations, error) {
 	imagePullAuths := &docker.AuthConfigurations{
-		map[string]docker.AuthConfiguration{"": {}}}
+		map[string]docker.AuthConfiguration{}}
 	if dockercfg != "" {
-		reader, err := os.Open(dockercfg)
-		if err != nil {
-			return nil, fmt.Errorf("Unable to open docker config file: %v\n", err)
-		}
-		if imagePullAuths, err = docker.NewAuthConfigurations(reader); err != nil {
-			return nil, fmt.Errorf("Unable to parse docker config file: %v\n", err)
-		}
-		if len(imagePullAuths.Configs) == 0 {
-			return nil, fmt.Errorf("No auths were found in the given dockercfg file\n")
+		for _, dcfgFile := range strings.Split(dockercfg, ",") {
```

Comment:
If we do end up with a custom impl (and not pflags) let's make sure the flag parsing stays in the cmd/image-inspector file (or package if it is a util).  I've been (mostly) trying to follow the Complete-Validate-Run scheme of Kube: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#command-implementation-conventions